### PR TITLE
Feature/add attribute

### PR DESCRIPTION
**Pull request by fdlk**
#### Checklist
- [x] Unit tests
- [x] Updated testplan

### CODE REVIEW

**Comment by Mark-de-Haan** [comment]
If I create an attribute and add it to an existing entity
the Sequence number I set is ignored, and the attribute is added at the end.

Editing the Sequence number after creation does set the order correctly

File: molgenis-data-rest/src/main/java/org/molgenis/data/rest/RestController.java [hunk 1]
```diff
@@ -735,7 +739,14 @@ public void delete(@PathVariable("entityName") String entityName, @PathVariable(
 		EntityType entityType = dataService.getEntityType(entityName);
 		Object id = getTypedValue(untypedId, entityType.getIdAttribute());
 
-		dataService.deleteById(entityName, id);
+		if (ATTRIBUTE_META_DATA.equals(entityName))
```
Comment (dennishendriksen) [inline molgenis-data-rest/src/main/java/org/molgenis/data/rest/RestController.java]: Metadata == data, the REST controller shouldn't make any entity specific exceptions.

File: molgenis-data-rest/src/main/java/org/molgenis/data/rest/RestController.java [hunk 2]
```diff
@@ -1056,13 +1068,21 @@ private void updateInternal(String entityName, String untypedId, Map<String, Obj
...
 
+	@Transactional
 	private void createInternal(String entityName, Map<String, Object> entityMap, HttpServletResponse response)
 	{
-		EntityType meta = dataService.getEntityType(entityName);
+		EntityType entityType = dataService.getEntityType(entityName);
+		Entity entity = this.restService.toEntity(entityType, entityMap);
 
-		Entity entity = this.restService.toEntity(meta, entityMap);
+		if (ATTRIBUTE_META_DATA.equals(entityName))
```
Comment (dennishendriksen) [inline molgenis-data-rest/src/main/java/org/molgenis/data/rest/RestController.java]: Metadata == data, the REST controller shouldn't make any entity specific exceptions.

File: molgenis-data-rest/src/main/java/org/molgenis/data/rest/v2/RestControllerV2.java [hunk 3]
```diff
@@ -205,7 +207,14 @@ public void deleteEntity(@PathVariable("entityName") String entityName, @PathVar
 		EntityType entityType = dataService.getEntityType(entityName);
 		Object id = getTypedValue(untypedId, entityType.getIdAttribute());
 
-		dataService.deleteById(entityName, id);
+		if (ATTRIBUTE_META_DATA.equals(entityName))
```
Comment (dennishendriksen) [inline molgenis-data-rest/src/main/java/org/molgenis/data/rest/v2/RestControllerV2.java]: Metadata == data, the REST controller shouldn't make any entity specific exceptions.

File: molgenis-data-rest/src/main/java/org/molgenis/data/rest/v2/RestControllerV2.java [hunk 4]
```diff
@@ -283,7 +292,14 @@ public EntityCollectionBatchCreateResponseBodyV2 createEntities(@PathVariable("e
 			final List<String> ids = new ArrayList<String>();
 
 			// Add all entities
-			this.dataService.add(entityName, entities.stream());
+			if (ATTRIBUTE_META_DATA.equals(entityName))
```
Comment (dennishendriksen) [inline molgenis-data-rest/src/main/java/org/molgenis/data/rest/v2/RestControllerV2.java]: Metadata == data, the REST controller shouldn't make any entity specific exceptions.

File: molgenis-data/src/main/java/org/molgenis/data/meta/MetaDataService.java [hunk 5]
```diff
@@ -267,4 +264,14 @@
...
+
+	/**
+	 * Performs an operation on each concrete child {@link EntityType} directly or indirectly extending a given
+	 * {@link EntityType}.
+	 * If the {@link EntityType} is concrete, will only perform the operation on the given {@link EntityType}.
+	 *
+	 * @param entityType the {@link EntityType} whose concrete child entities will be consumed
+	 * @param consumer   the operation to perform
+	 */
+	void forEachConcreteChild(EntityType entityType, Consumer<EntityType> consumer);
```
Comment (dennishendriksen) [inline molgenis-data/src/main/java/org/molgenis/data/meta/MetaDataService.java]: Why not request a stream of entity types and call forEach?

**Comment by Mark-de-Haan** [comment]
Observed on your branch: #5426 #5424

File: molgenis-data-rest/src/main/java/org/molgenis/data/rest/v2/RestControllerV2.java (hunk 4 above)
Comment (fdlk) [inline molgenis-data-rest/src/main/java/org/molgenis/data/rest/v2/RestControllerV2.java]: But let's make it work before we make it pretty :)

**Comment by fdlk** [comment]
test this please

**Comment by Mark-de-Haan** [comment]
When I try to delete a package:
`Error! Could not resolve dependencies of items [org_molgenis_test_TypeTest, sys_FileMeta]. Are there circular dependencies?`

**Comment by Mark-de-Haan** [comment]
When adding an attribute, the Sequence order is not applied

**Comment by jjettenn** [comment]
"When I try to delete the org_molgenis_test package:" cannot reproduce error.
Added bug: Delete package does not results in refreshing entity select box in "Data Explorer" #5452

**Pull request merged**